Sources/NIOConcurrencyHelpers/lock.swift

```diff
@@ -160,6 +204,9 @@ public final class ConditionLock<T: Equatable> {
     public func lock(whenValue wantedValue: T, timeoutSeconds: Double) -> Bool {
         precondition(timeoutSeconds >= 0)
 
+#if os(Windows)
+          fatalError()
```

Comment:
This needs to be ported over, but cannot be accurate as `SleepConditionVariableSRW` takes a relative time rather than absolute time.  It also would have a +/-10ns time-frame as timers on Windows have a 10ns granularity.

Reply:
I don't think I'm overly worried about timer granularity, so I don't really consider that to be a problem. I'm not sure I follow the relative vs absolute time concerns though.

Reply:
Just that the sleep will not be exactly precise.  I will have some overheads due to adjustments for the remaining sleep just in case of a spurious wakeup.